tests/ignite/handlers/test_ema_handler.py

```diff
@@ -0,0 +1,294 @@
+import os
+from typing import Any, Callable, Union
+
+import pytest
+import torch
+import torch.nn as nn
+from torch.nn.parallel import DataParallel, DistributedDataParallel
+from torch.utils.data import DataLoader, TensorDataset
+
+import ignite.distributed as idist
+from ignite.engine import Engine, Events
+from ignite.handlers import EMAHandler
+
+
+def _get_dummy_dataloader() -> DataLoader:
+    """Get a dummy data loader with length of 2"""
+    x = torch.randint(0, 10, (2, 2))
+    data_set = TensorDataset(x)
+    data_loader = DataLoader(data_set, batch_size=1)
+    return data_loader
+
+
+def _get_dummy_model() -> nn.Module:
+    model = nn.Linear(2, 1, bias=False)
+    model.weight.data.fill_(1)
+    return model
+
+
+def _get_dummy_step_fn(model: Union[nn.Module, DataParallel, DistributedDataParallel]) -> Callable:
+    """Get a dummy step function, given model is a (wrapper of) dummy model returned from _get_dummy_model"""
+
+    def step_fn(engine, batch):
+        """Increment the weight by 1 at each iteration"""
+        model.weight.data.add_(1)
+        return 0
+
+    return step_fn
+
+
+@pytest.mark.parametrize("momentum_end", [-1, 2])
+def test_ema_invalid_momentum_end(momentum_end):
+    with pytest.raises(ValueError, match="Invalid momentum_end"):
+        model = _get_dummy_model()
+        EMAHandler(model, momentum_end=momentum_end)
+
+
+@pytest.mark.parametrize("momentum_start", [-1, 2])
+def test_ema_invalid_momentum_start(momentum_start):
+    with pytest.raises(ValueError, match="Invalid momentum_start"):
+        model = _get_dummy_model()
+        EMAHandler(model, momentum_start=momentum_start)
+
+
+def test_ema_invalid_momentum_start_end():
+    """Test momentum_end > momentum_start"""
+    momentum_end = 0.001
+    momentum_start = 0.1
+    with pytest.raises(ValueError, match="momentum_start should be less than or equal to momentum_end"):
+        model = _get_dummy_model()
+        EMAHandler(model, momentum_start=momentum_start, momentum_end=momentum_end)
+
+
+def test_ema_invalid_model():
+    with pytest.raises(ValueError, match="model should be an instance of nn.Module or its subclasses"):
+        model = "Invalid Model"
+        EMAHandler(model)  # type: ignore
+
+
+@pytest.mark.distributed
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="Skip if no GPU")
+def test_ema_ema_model():
```

Comment:
```suggestion
def test_ema_ema_model_on_cuda():
```
We have a convention to add `cuda` to make the test run on GPUs